src/edu/stanford/nlp/parser/dvparser/DVParser.java

```diff
@@ -154,7 +154,7 @@ public void train(List<Tree> sentences, IdentityHashMap<Tree, byte[]> compressed
         // the list
         int startTree = batch * op.trainOptions.batchSize;
         int endTree = (batch + 1) * op.trainOptions.batchSize;
-        if (endTree + op.trainOptions.batchSize > shuffledSentences.size()) {
+        if (endTree > shuffledSentences.size()) {
```

Comment:
I *think* that the subsequent call to `shuffledSentences.subList(startTree, endTree)` will keep the last sentences from being repeated.  However, it certainly is ugly written the way it is, so I will merge this change.